### ISSUE TITLE
Staking process and overview ideas

### ISSUE THREAD

**Issue by Tbaut** [polkadot-js/apps#958]
I thought about a step by step process that would let users validate/nominate following a simple process, with guidance thanks to all the (?) along the way, and with a limited amount of signing modals. The goal is to :
1. Setup all you need to validate/nominate in a couple steps and sign 1 extrinsic (allow me to dream :drooling_face: ).
1. Present in the `Account Actions` only the accounts involved, and group them.
1. Have a few call to actions to start validating/nominating right away.

There are limitations currently on the API side e.g the inability to set the controller, session key, bounded amount, reward destination and commission, all in 1 extrinsic). ping @kianenigma
There might be other technical limitations, let me know.

I want to first see what you think, know if that's something we could eventually do, before getting out and validate the ideas or even get some more by talking to the validators.

Few call to actions to start with and an empty page:
![image](https://user-images.githubusercontent.com/33178835/55909948-7fb55100-5bdd-11e9-83d8-21d92b33d15e.png)

**Validation setup step 1/3:**
Start with the `Stash`, the bounding amount and the `Session` account.
![image](https://user-images.githubusercontent.com/33178835/55910033-b1c6b300-5bdd-11e9-8b0e-61a180b04f7e.png)
**Validation setup step 2/3:**
Setup the `Controller` and all the params attached to it. Note that renaming the "payment preferences" into "commission" and expressing it in percentage (with a slider) seems to me easier to grasp.
![image](https://user-images.githubusercontent.com/33178835/55910102-dd499d80-5bdd-11e9-845a-5c9d15f9b7c3.png)

**Validation example step 3/3:**
Overview, no more (?) as everything should have been covered before.
![image](https://user-images.githubusercontent.com/33178835/55910128-edfa1380-5bdd-11e9-91ec-f05aa3d9bbc3.png)

**Accounts Actions overview:**
Group the accounts together without size difference. Show all info.
Not represented on this screen, users could edit any param by clicking on an edit button that would appear a mouse over next to each parameter or account. e.g change controller/session/bounded/commission... Depending on the parameter to edit, the user would have to sign with the `stash` or `controller` account.
![image](https://user-images.githubusercontent.com/33178835/55910848-affdef00-5bdf-11e9-8292-8f6ed6635c88.png)

Additional ideas:
- On the validation/nomination setup, show at the bottom of the modal an indication that the user is at step x/3
- on the `Account Actions` screen, to keep it a little simpler we could think about hiding some of the parameters in a "show/hide details" slide-down. e.g all except "bounded amount".

**Comment by jacogr** [comment]
First off, the fact that there are multiple extrinsics, shouldn't be a show-stopper. We have the ability to send and resume with some magic. (So optional, and generally the node calls doesn't need to bend to UX will)

I believe this issue should be broken into parts. But first off there are a bunch of issues that should be addressed before we get here, by making sure we have all the functionality: set controller (https://github.com/polkadot-js/apps/issues/950), set prefs (https://github.com/polkadot-js/apps/issues/945), unbond/withdraw (https://github.com/polkadot-js/apps/issues/859)  and optionally just some better guidance, since it is confusing around eras and sessions (https://github.com/polkadot-js/apps/issues/861)

(All the above it in the queue and actually simple technically, i.e. hookup a TxButton with the params, but more difficult to get the flow right UX-wise since some are actually just more buttons on-screen and others feel counter-intuitive.)

After that will come back, the "combine all" is something that I have on my list, and it almost a starting point to see how it plays out. The "first use" is also something very interesting here, since it is actually something that we want to expand to a couple of places.

TL;DR - Going to turn this into a meta issue, logistically it is not today.

**Comment by Kianenigma** [comment]
@Tbaut Generally a much better approach for staking. Some initial thoughts: 

- I totally agree that the account action page should go from "showing all accounts with two buttons stick to their forehead" to just two buttons up top and just the list of accounts that are actually doing stuff with the ability to chill. Note that staking is (at least for now) kinda one-way bond; there is no way back, the `unbond()` call just withdraws the funds, not releasing the accounts.

- It is not impossible to make the idealistic call that does everything all at once but unless if it is just for testing, I have to run it by Gavin quickly and then get it shipped since it might have some implications that we, from the perspective of UX, do not take into account. I am still in favor of doing these tricks in the UI (e.g. validating always needs SetKey so you really don't need to ask the user again -- you can put a checkbox there with the title `automatically set stash/ctrl key` that if checked, does not ask the user to sign again in a new box and sets the session key under the hood) (ok I just realized the idea in the previous parenthesis is quite useful actually :D).

- On a smaller scale and note, commission ()originally called `validator_payment` or `off_the_table_payment`) is not a ratio or percentage by nature (just a normal `Balance`) so keep that in mind. You can make it a ratio on the front side w.r.t. the total bonded value.

This could be a good end goal but as @jacogr said probably there are more basic steps missing; Though I wonder if they are all worth fixing if they are to be changed.

**Comment by jacogr** [comment]
Limited resources on this side, so existing additions take priority above all else. Basic is better than ideal, so will work towards that, but it is not going to happen overnight.
